crates/matrix-sdk-common/src/js_tracing.rs

```diff
@@ -199,9 +202,130 @@ fn write_message_to_console(level: Level, message: &JsValue) {
     };
 }
 
+/// An implementation of [`FormatEvent`] which formats events in a sensible way
+/// for sending events to the JS console.
+#[derive(Debug, Default)]
+pub struct JsEventFormatter {}
```

Comment:
```suggestion
#[non_exhaustive]
pub struct JsEventFormatter {}
```

If we want to enforce that `::new()` is the way to create an instance.